src/plonk/prover.rs

```diff
@@ -451,13 +455,9 @@ impl<C: CurveAffine> Proof<C> {
             .chain(permutation_product_inv_evals.iter())
             .chain(permutation_evals.iter().flat_map(|evals| evals.iter()))
         {
-            transcript_scalar.absorb(*eval);
+            transcript.absorb_scalar(*eval);
         }
 
-        let transcript_scalar_point =
-            C::Base::from_bytes(&(transcript_scalar.squeeze()).to_bytes()).unwrap();
-        transcript.absorb(transcript_scalar_point);
```

Comment:
This is the squeeze I'm referring to above, which is now deferred until after `f_commitment` is absorbed. If we conditionally squeezed inside `Transcript::absorb_point`, we'd retain the current behaviour for this specific case.

Reply:
(And similarly in `Transcript::absorb_base`)

Reply:
Moving an absorption earlier is safe, i.e. always preserves knowledge soundness (because all squeezed challenges will depend at least on everything they depended on before). Moving a squeeze earlier is not safe in general.